### DISCUSSION TITLE
umxTwinify

### DISCUSSION THREAD

**Discussion by mcneale** [tbates/umx#166] [Ideas]
Given an SEM for an individual, automatically generate a twin version.  The model should be specified in path syntax, then exogenous latent variables can be discovered by examining their corresponding row of the A matrix, which should have no free parameter or non-zero entries.  This latent variable then gets ACE variables specifically pointing to it (3 more columns of the A and S matrices), repeat for all latent variables.  Next, duplicate the model for twin 2, rename the variables, and put in covariance paths between the A's and C's.  Could be done separately for MZ and DZ, umxSuperModel them, and Robert's your mother's brother eh?

**Comment by tbates** [comment]
your wish is my command :-)

```R
paths = c(
   umxPath(v1m0 = c("a1", 'c1', "e1")),
   umxPath(means = c("wt")),
   umxPath(c("a1", 'c1', "e1"), to = "wt", values=.2)
)
# 2. Make a twin model from the paths for one person
m1 = umxTwinMaker("test", paths, mzData = mzData, dzData= dzData)


```
